rendezvous uTP connection API

vinipsmaker opened the pull request:
Necessary to use the hole punched UDP sockets.

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/maidsafe/crust/366)

<!-- Reviewable:end -->

maidsafe-highfive commented:
r? @canndrew

(maidsafe_highfive has picked a reviewer for you, use r? to override)

vinipsmaker commented:
r? @inetic

vinipsmaker commented:
Now that [commit https://github.com/maidsafe/crust/commit/e3bf5bd6ee96fb180741305587fbdada9642d3a9](https://github.com/maidsafe/crust/commit/e3bf5bd6ee96fb180741305587fbdada9642d3a9) is in place, PR is ready for review/merge.

vinipsmaker committed:
add test to ensure CBOR can be used to meet our demands

And looks like I was wrong about the way CBOR crate behaved.

Closes #374.

vinipsmaker committed:
readd loop that was removed in commit e3bf5bd6ee96fb180741305587fbdada9642d3a9

Also added comment to avoid this loop being removed again.

vinipsmaker committed:
reduce `service::test::rendezvous_connection` to ease debugging

`service::test::rendezvous_connection` shouldn't call `accept` anymore
and should be easy to find any issue if this test show any.

vinipsmaker committed:
unignore rendezvous connection test

inetic commented on src/service.rs:
```diff
@@ -642,9 +642,6 @@ mod test {
 
-    // Temporarily ignored this test, should be fixed once a pending PR
-    // to https://github.com/maidsafe/rust-utp/pull/5 gets merged.
-    #[ignore]
```
:)
